pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -346,57 +346,66 @@ func (c *NPLController) handleRemovePod(key string, obj interface{}) error {
 	if obj != nil {
 		pod := obj.(*corev1.Pod)
 		if _, exists := pod.Annotations[NPLAnnotationKey]; exists {
-			newPod := pod.DeepCopy()
-			removePodAnnotation(newPod)
-			return c.updatePodAnnotation(newPod)
+			return CleanupNPLAnnotationForPod(c.kubeClient, *pod)
```

Comment:
CleanupNPLAnnotationForPod has the same check as L348

Reply:
Yeah, removed it from here, not required.